app/services/badge_granter.rb

```diff
@@ -450,4 +450,10 @@ def self.send_notification(user_id, username, locale, badge)
     notification
   end
 
+  def self.suppress_notification?(badge, granted_at, skip_new_user_tips)
```

Comment:
I could not make this class method private, unfortunately because we call it from an instance method `grant`.